cpp/include/cugraph_c/graph_functions.h

```diff
@@ -324,6 +324,24 @@ cugraph_error_code_t cugraph_degrees(const cugraph_resource_handle_t* handle,
                                      cugraph_degrees_result_t** result,
                                      cugraph_error_t** error);
 
+/**
+ * @brief       Construct the edge list from the graph view object.
+ *
+ * @param [in]  handle              Handle for accessing resources
+ * @param [in]  graph               Graph to operate on
+ * @param [in]  do_expensive_check  A flag to run expensive checks for input arguments (if set to
+ * true)
+ * @param [out] result              Opaque pointer to induced subgraph result
+ * @param [out] error               Pointer to an error object storing details of any error.  Will
+ *                                  be populated if error code is not CUGRAPH_SUCCESS
+ * @return error code
+ */
+cugraph_error_code_t cugraph_decompress_to_edgelist(const cugraph_resource_handle_t* handle,
+                                                    cugraph_graph_t* graph,
+                                                    bool_t do_expensive_check,
+                                                    cugraph_induced_subgraph_result_t** result,
```

Comment:
Just a suggestion, but `cugraph_induced_subgraph_result_t` sounds a bit of misnomer here. We may used this data structure to just store edge list from the induced subgraph algorithm in the past, but if we wish to use this for other algorithms as well, we may better rename this now or sometime in the future.

Reply:
Agreed.  I would suggest `cugraph_edgelist_t`.  I have a draft PR (that I need to discard and start over with) that was going to create a `cugraph_edgelist_t` for the C API for graph construction.

I'd suggest the following:
* Replicate the `cugraph_induced_subgraph_result_t` type and C API accessor functions to call it `cugraph_edgelist_t` instead
* Mark the induced subgraph variations as deprecated
* Within the C API implementation (`src/c_api`)  you can either rename the `cugraph::c_api::cugraph_induced_subgraph_result_t` to `cugraph::c_api::cugraph_edgelist_t` and rename all references accordingly, or you can leave it using the legacy version and I'll fix it in 25.02 when I finish the C API graph construction updates.